staging/src/k8s.io/apiserver/pkg/storage/value/metrics.go

```diff
@@ -46,9 +96,27 @@ func RegisterMetrics() {
 	})
 }
 
-func RecordTransformation(transformationType string, start time.Time) {
-	since := sinceInMicroseconds(start)
-	transformerLatencies.WithLabelValues(transformationType).Observe(float64(since))
+func RecordTransformation(transformationType string, start time.Time, err error) {
+	if err == nil {
+		since := sinceInMicroseconds(start)
+		transformerLatencies.WithLabelValues(transformationType).Observe(float64(since))
+		transformerOperationsTotal.WithLabelValues(transformationType).Inc()
+	} else {
+		transformerFailuresTotal.WithLabelValues(transformationType).Inc()
+	}
+}
+
+func RecordCacheMiss() {
+	envelopeTransformationCacheMissTotal.Inc()
+}
+
+func RecordDataKeyGeneration(start time.Time, err error) {
+	if err == nil {
```

Comment:
Keep error flow indented:
```
	if err != nil {
		dataKeyGenerationFailuresTotal.Inc()
	}
	since := sinceInMicroseconds(start)
	dataKeyGenerationLatencies.Observe(float64(since))
```

Same in RecordTransformation

Reply:
Done.